src/request_parser.rs

```diff
@@ -1,6 +1,34 @@
 //! Methods for retrieving swagger-related information from an HTTP request.
 use hyper::Request;
 
+/// A macro for joining together two or more RequestParsers to create a struct that implements
+/// RequestParser with a function parse_operation_id that matches hyper requests against the different
+/// RequestParsers in turn until it gets a match (or returns an error if none match)
+#[macro_export]
+macro_rules! request_parser_joiner {
+    ($name:ident ,$($T:ty), *) => {
+        struct $name;
+
+        impl RequestParser for $name {
+            fn parse_operation_id(request: &hyper::Request) -> Result<&'static str, ()> {
+                inner!(request, $($T), *)
+            }
+        }
+    };
+}
+
+/// This macro should only be used by the request_parser_joiner macro
+#[macro_export]
+macro_rules! inner {
```

Comment:
Can you rename this?

Currently, any 2015 project which imports this will get a `inner!` macro defined which may conflict with another macro.

Suggest: `__impl_request_parser_joiner` to make it clear, a) what it does, b) what it's internal implementation detail only. This follows the style suggested in https://doc.rust-lang.org/edition-guide/rust-2018/macros/macro-changes.html.

Suggest that you also add `#[doc(hidden)]` so it's hidden from the public documentation - which means it's less likely to get used.